cmds/webboot/webboot.go

```diff
@@ -0,0 +1,307 @@
+		return err
+	}
+	verbose("iso is downloaded")
+
+	return nil
+}
+
+// getKernel is to find the kernel of a ISO. 
+func getKernel(u *ISO) error {
+
+	/*
+	* TODO: code this function after FindDevice is ready
+	*/
+	return nil
+}
+
+// isoMenu is to  build the iso menu page and return the iso path
+// and kernel path of which user choose.
+func isoMenu(isos []ISO) error {
+
+	var entries []menus.Entry
+	for _, iso := range isos {
+		entries = append(entries, iso)
+	}
+	var downloadOption DownloadOption
+	var downloadByLinkOption DownloadByLinkOption
+	entries = append(entries, downloadOption, downloadByLinkOption)
+
+	menuTitle := "ISO Menu"
+	introwords := "Choose an iso you want to boot (hit enter to choose the default - 1 is the default option) >"
+	location := 0
+	entry, err := menus.DisplayMenu(menuTitle, introwords, location, entries)
+
+	// if index is exceed the length of entries or less than 0 means something wrong happened
+	if err != nil {
+		return err
+	}
+	err = entry.Do()
+	return err
+	return nil
+}
+
+// linkOpen returns an io.ReadCloser that holds the content of the URL
+func linkOpen(URL string) (io.ReadCloser, error) {
+	switch {
+	case strings.HasPrefix(URL, "file://"):
+		return os.Open(URL[7:])
+	case strings.HasPrefix(URL, "http://"), strings.HasPrefix(URL, "https://"):
+		resp, err := http.Get(URL)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("HTTP Get failed: %v", resp.StatusCode)
+		}
+		return resp.Body, nil
+	}
+	return nil, fmt.Errorf("%q: linkopen only supports file://, https://, and http:// schemes", URL)
+}
+
+// downloadIso will download a iso from URL and save it under savePath directory
+func downloadIso(isoPath, URL string) error {
+	verbose("Should begin to download here..., download url is %s\n", URL)
+	introwords := "Please input the name of the interface:"
+	wid := 100
+	ht := 1
+	interfaceNameCheckFunc := func(input string) (bool, string) {
+		if input[0] == 'e' || input[0] == 'w' {
+			return true, ""
+		}
+		return false, "not a valid interface name"
+	}
+	ifName, err := menus.NewInputWindow(introwords, wid, ht, interfaceNameCheckFunc)
+	if err != nil {
+		return err
+	}
+	if *ipv4 || *ipv6 {
+		dhclient.Request(ifName, *timeout, *retry, *v, *ipv4, *ipv6)
+	}
+	isoReader, err := linkOpen(URL)
+	if err != nil {
+		return err
+	}
+	defer isoReader.Close()
+	isofile, err := os.Create(isoPath)
+	if err != nil {
+		return err
+	}
+	if _, err := io.Copy(isofile, isoReader); err != nil {
+		return fmt.Errorf("Error copying to persistent memory device: %v", err)
+	}
+	if err = isofile.Close(); err != nil {
+		return fmt.Errorf("Error closing %s: %v", isoPath, err)
+	}
+
+	verbose("downloaded isoPath is %s", isoPath)
+	return nil
+}
+
+func main() {
+	flag.Parse()
+
+	if *v {
+		verbose = log.Printf
+	}
+
+	var isos []ISO
+
+	// If isoDir is not given, that means we need use findDevice to find the USB disk
```

Comment:
You can just add a comment here saying TODO: support finding ISOs from bootable device